config/jobs/README.md

```diff
@@ -1,9 +1,213 @@
+
+Please see [ProwJob docs](/prow/jobs.md) for more info
+
+### Job Images
+
+Where possible, we prefer that jobs use images that are pinned to a specific
+version, containing only what is needed.
+
+Some good examples include:
+
+- [pull-release-unit] uses `golang:1.12` to run `go test ./...`
+- [pull-release-notes-lint] uses `node:11` to run `npm ci && npm lint`
+- [pull-org-test-all] uses `launcher.gcr.io/google/bazel:0.26.0` to run `bazel test //...`
+
+Many jobs use `gcr.io/k8s-testimages/foo` images that are built from source in
+[`images/`]. Some of these have evolved organically, with way more dependencies
+than needed, and will be periodically bumped by PRs. These are sources of
+technical debt that are often not very well maintained. Use at your own risk,
+eg:
+
+- [pull-community-verify] uses `gcr.io/k8s-testimages/gcloud-in-go:v20190125`
+  to run `make verify`, which ends up invoking some `bash` scripts, which use
+  commands like `git` and `go`. The `gcloud` dependency is not needed at all.
+- [periodic-kubernetes-e2e-packages-pushed] uses `gcr.io/k8s-testimages/kubekins:latest-master`
+  to run `./tests/e2e/packages/verify_packages_published.sh` which ends up
+  running `apt-get` and `yum` commands. Perhaps a `debian` image would be
+  better.
+
+## Job Presets
+
+Many jobs require common secrets, configmaps, environment variables, volumes,
+etc. Prow supports the use of presets to define and patch these in. Some are
+defined centrally in [`prow/config.yaml`], while others can be defined in
+files here. eg:
+
+- [`preset-service-account: "true"`] ensures the prowjob has a GCP service
+  account in a well known location, with well known env vars pointing to it.
+- [`preset-pull-kubernetes-e2e: "true"`] sets environment variables to make
+  kubernetes e2e tests less susceptible to flakes
+- [`preset-aws-credentials: "true"`] ensures the prowjob has AWS credentials
+  for kops tests in a well known location, with an env var pointint to it
+
+## Job Examples
+
+A presubmit job named "pull-community-verify" that will run against all PRs to
+kuberentes/community's master branch. It will run `make verify` in a checkout
+of kubernetes/community at the PR's HEAD. It will report back to the PR via a 
+status context named `pull-kubernetes-community`. Its logs and results are going 
+to end up in GCS under `kubernetes-jenkins/pr-logs/pull/community`. It's using
+the `gcloud-in-go` image, which isn't recommended, but works for now.
+
+```yaml
+presubmits:
+  kubernetes/community:
+  - name: pull-community-verify  # convention: (job type)-(repo name)-(suite name)
+    branches:
+    - master
+    decorate: true
+    always_run: true
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/gcloud-in-go:v20190125-cc5d6ecff3
+        args:
+        - "--repo=k8s.io/$(REPO_NAME)=$(PULL_REFS)"
+        - "--root=/go/src"
+        - "--upload=gs://kubernetes-jenkins/pr-logs"
+        - --scenario=execute
```

Comment:
These are `bootstrap.py` args which are not needed with pod utility decoration (which is enabled above). I think this would actually result in duplicate uploads to GCS from `bootstrap.py` and `initupload`+`sidecar` for some surprise behavior 🙃

Reply:
I was trying to find a good example of something that does more than just a single language-specific command (`go test` etc.), and this was the closest I could find.  Realized the same thing in looking at it.  This is a thing that actually runs today, how about I break fixing this job (and this example) into an issue.  I'm also open to suggestion of a different job

Reply:
I opened https://github.com/kubernetes/test-infra/issues/13118

Reply:
updating with contents of https://github.com/kubernetes/test-infra/pull/13126

Reply:
(which don't work, ok...)